addons/website/static/src/snippets/s_dynamic_snippet/dynamic_snippet.js

```diff
@@ -94,6 +117,8 @@ export class DynamicSnippet extends Interaction {
 
     async fetchData() {
         if (this.isConfigComplete()) {
+            const remaining = this.totalToFetch - this.fetchedData.length;
+            this.limit = Math.min(this.chunkSize, remaining);
```

Comment:
What if I set Fetched Elements = 20, chunkSize is by default 16, shouldn't we use Fetched Elements as chunk size?